### PR TITLE
Parser tuned.conf

### PR DESCRIPTION
**Pull request by vishwanathjadhav**
Added a parser to parse configuration file `/etc/tuned.conf` of tuned

### CODE REVIEW

File: insights/parsers/tuned_conf.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+from .. import parser, Parser, get_active_lines
+from insights.specs import Specs
+from insights.contrib.ConfigParser import NoSectionError, NoOptionError
+
+
+@parser(Specs.tuned_conf)
+class TunedConfIni(Parser):
+    """This class parses the ``/etc/tuned.conf`` file using the
+    ``IniConfigFile`` base parser.
```
Comment (xiangce) [inline insights/parsers/tuned_conf.py]: Are you trying to use the `IniConfigFile` parser as its base class? Seems by doing this, this parser can be simplified very much.

File: insights/parsers/tuned_conf.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+"""
+This module contains following parser:
```
Comment (xiangce) [inline insights/parsers/tuned_conf.py]: You may want to add a title for this file:
```
'''
Title XXX
=========
'''
```

File: insights/specs/__init__.py [hunk 3]
```diff
@@ -491,6 +491,7 @@ class Specs(SpecSet):
     uptime = RegistryPoint()
     usr_journald_conf_d = RegistryPoint(multi_output=True)
     vdsm_conf = RegistryPoint()
+    tuned_conf = RegistryPoint()
```
Comment (xiangce) [inline insights/specs/__init__.py]: Better insert this spec as per the alphabet order.

File: insights/parsers/tuned_conf.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+TunedConfIni - file ``/etc/tuned.conf``
+---------------------------------------
+"""
+
+from .. import parser, Parser, get_active_lines
```
Comment (xiangce) [inline insights/parsers/tuned_conf.py]: Better uniform the usage of the `from` path in one single python parser file.
```
from insights.parsers import parser, Parser, get_active_lines
```

File: insights/parsers/tuned_conf.py (hunk 1 above)
Comment (vishwanathjadhav) [reply insights/parsers/tuned_conf.py]: Current INI parser is not able to handle one corner case "# enabled=False" "<single space>enabled=False", throws error.

File: insights/parsers/tuned_conf.py (hunk 2 above)
Comment (vishwanathjadhav) [reply insights/parsers/tuned_conf.py]: Done

File: insights/parsers/tuned_conf.py (hunk 4 above)
Comment (vishwanathjadhav) [reply insights/parsers/tuned_conf.py]: Done

File: insights/specs/__init__.py (hunk 3 above)
Comment (vishwanathjadhav) [reply insights/specs/__init__.py]: Done

File: insights/parsers/tuned_conf.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    def parse_content(self, content, allow_no_value=True):
+        content = get_active_lines(content)
+
+        super(IniConfigFile, self).parse_content(content)
+        config = RawConfigParser(allow_no_value=allow_no_value)
+        fp = io.StringIO(u"\n".join(content))
+        config.readfp(fp, filename=self.file_name)
+        self.data = config
+    pass
```
Comment (xiangce) [inline insights/parsers/tuned_conf.py]: This `pass` can be deleted.

File: insights/parsers/tuned_conf.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        >>> tuned_obj.getboolean('NetTuning', 'enabled') == False
+        True
+        >>> sorted(tuned_obj.sections())==sorted(['CPUMonitor', 'NetTuning'])
+        True
+    """
+
+    def parse_content(self, content, allow_no_value=True):
+        content = get_active_lines(content)
+
+        super(IniConfigFile, self).parse_content(content)
```
Comment (xiangce) [inline insights/parsers/tuned_conf.py]: The better way is to call the super of `TunedConfIni` but not `IniConfigFile`

```
super(TunedConfIni, self).parse_content(content)

```
Then the following lines can be removed.

Comment (vishwanathjadhav) [reply insights/parsers/tuned_conf.py]: Done

File: insights/parsers/tuned_conf.py (hunk 5 above)
Comment (vishwanathjadhav) [reply insights/parsers/tuned_conf.py]: Done

**Pull request merged**